### PR TITLE
Register

### PR DESCRIPTION
**Pull request by joredudiaz**
## Summary

- Register view
- Show by console the data send

## Jira Story

https://trello.com/c/1Cj8xwsr
https://trello.com/c/VxaRu41O

## Screenshot

register

### CODE REVIEW

File: wolox-books/src/app/screens/user/components/register/register.component.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<form [formGroup]="formGroup" (ngSubmit)="request(formGroup.value)" class="form">
```
Comment (betoobernal) [inline wolox-books/src/app/screens/user/components/register/register.component.html]: inside of request method read `formGroup.value`

File: wolox-books/src/app/screens/user/components/register/register.component.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  <label for="firstname" class="label">Nombre</label>
+  <input type="text" class="input-text" formControlName="first_name">
+  <div class="error" *ngIf="!formGroup.controls.first_name.valid && formGroup.controls.first_name.touched">This field is required.</div>
```
Comment (betoobernal) [inline wolox-books/src/app/screens/user/components/register/register.component.html]: use get accessor in the class and that will do the HTML more readable.

File: wolox-books/src/app/screens/user/components/register/register.component.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  <label for="lastname" class="label">Apellido</label>
+  <input type="text" class="input-text" formControlName="last_name">
+  <div class="error" *ngIf="!formGroup.controls.last_name.valid && formGroup.controls.last_name.touched">This field is required.</div>
```
Comment (betoobernal) [inline wolox-books/src/app/screens/user/components/register/register.component.html]: same

File: wolox-books/src/app/screens/user/components/register/register.component.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  <label for="email" class="label">Email</label>
+  <input type="text" class="input-text" formControlName="email">
+  <div class="error" *ngIf="!formGroup.controls.email.valid && formGroup.controls.email.touched">You must specify an email valid.</div>
```
Comment (betoobernal) [inline wolox-books/src/app/screens/user/components/register/register.component.html]: and the rest of them. 😅

File: wolox-books/src/app/screens/user/components/register/register.component.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss']
+})
+export class RegisterComponent implements OnInit {
+
+  formGroup: FormGroup;
+
+  data: any;
+
+  constructor(private formBuilder: FormBuilder) {
```
Comment (betoobernal) [inline wolox-books/src/app/screens/user/components/register/register.component.ts]: if you don't mutate these objects they should be `readonly`
```suggestion
  constructor(private readonly formBuilder: FormBuilder) {
```

File: wolox-books/src/app/screens/user/components/register/register.component.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    this.formGroup = formBuilder.group({
+      'first_name': [null, Validators.required],
+      'last_name': [null, Validators.required],
+      'email': [null, Validators.compose([Validators.required, Validators.email])],
+      'password': [null, Validators.compose([Validators.required, Validators.pattern('(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{8,}')])],
```
Comment (betoobernal) [inline wolox-books/src/app/screens/user/components/register/register.component.ts]:
```suggestion
      'email': [null, [Validators.required, Validators.email]],
      'password': [null, [Validators.required, Validators.pattern('(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{8,}')]],
```

Angular inside call `Validators.compose(...)` when this array is passed.

File: wolox-books/src/app/screens/user/user.component.scss [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  display: flex;
+  flex-direction: column;
+  padding: 20px 25px 24px;
+  width: 300px;
+}
+
+.logo {
+  align-self: center;
+  height: 70px;
+  width: 212.42px;
```
Comment (betoobernal) [inline wolox-books/src/app/screens/user/user.component.scss:15]: 🤔

File: wolox-books/src/fonts.scss [hunk 8]
```diff
@@ -0,0 +1 @@
+@import url('https://fonts.googleapis.com/css2?family=Roboto&display=swap')
```
Comment (betoobernal) [inline wolox-books/src/fonts.scss]: add extra space

File: wolox-books/src/styles.scss [hunk 9]
```diff
@@ -1 +1,68 @@
 /* You can add global styles to this file, and also import other style files */
+%font-shared {
```
Comment (betoobernal) [inline wolox-books/src/styles.scss:2]: nice 🎉

File: wolox-books/src/styles.scss [hunk 10]
```diff
@@ -1 +1,68 @@
...
+  color: $white;
+  height: 32px;
+  margin-top: 24px;
+
+  &:hover {
+    cursor: pointer;
+  }
+
+  &:active {
+    background-color: #a0bf67;
```
Comment (betoobernal) [inline wolox-books/src/styles.scss]: use a variable

File: wolox-books/src/styles.scss [hunk 11]
```diff
@@ -1 +1,68 @@
...
+  justify-content: center;
+
+  &:hover {
+    color: $earls-green;
+  }
+
+  &:active {
+    border: solid #a0bf67 1px;
+    color: #a0bf67;
+  }
```
Comment (betoobernal) [inline wolox-books/src/styles.scss]: same

File: wolox-books/src/styles.scss (hunk 10 above)
Comment (joredudiaz) [reply wolox-books/src/styles.scss]: shangos I forgot the variable 😞

File: wolox-books/src/app/screens/user/components/register/register.component.html [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+  <div class="error" *ngIf="!lastName.valid && lastName.touched">This field is required.</div>
+  <label for="email" class="label">Email</label>
+  <input type="text" class="input-text" formControlName="email">
+  <div class="error" *ngIf="!email.valid && email.touched">You must specify an email valid.</div>
+  <label for="password" class="label">Password</label>
+  <input type="password" class="input-text" formControlName="password">
+  <div class="error" *ngIf="!password.valid && password.touched">This field is required with at least A-Z, a-z and 0-9 characters.</div>
+  <label for="password-confirm" class="label">Confirmaci&oacute;n de Password</label>
+  <input type="password" class="input-text" formControlName="password_confirmation">
+  <div class="error" *ngIf="equalTo(formGroup) && passwordConfirmation.touched">This field is required and equal to password.</div>
```
Comment (betoobernal) [inline wolox-books/src/app/screens/user/components/register/register.component.html]: `equalTo` as a get accessor. Pass `formGroup` as param is needless, yo can get it by reference.

```suggestion
  <div class="error" *ngIf="equalTo && passwordConfirmation.touched">This field is required and equal to password.</div>
```

**Review by betoobernal — APPROVED** [review]

**Pull request merged**